TriblerGUI/widgets/trustpage/js/drawing.js

```diff
@@ -119,21 +171,63 @@ function drawLinks(svg, data) {
     links.append("line")
         .attr("class", "link-source")
         .attr("stroke-width", function (d) {
-            return getStrokeWidth(d, data)
+            return getStrokeWidth(d, data);
+        })
+        .style("stroke", config.link.colorLinkSource)
+        .on("mouseover", function (d) {
+            mouseInEdge(d, hoverInfoLabel);
         })
-        .style("stroke", config.link.colorLinkSource);
+        .on("mouseout", function () {
+            hoverInfoLabel.style("opacity", config.node.hoverLabel.opacityOnNode);
+        });
 
     links.append("line")
         .attr("class", "link-target")
         .attr("stroke-width", function (d) {
-            return getStrokeWidth(d, data)
+            return getStrokeWidth(d, data);
+        })
+        .style("stroke", config.link.colorLinkTarget)
+        .on("mouseover", function (d) {
+            mouseInEdge(d, hoverInfoLabel);
         })
-        .style("stroke", config.link.colorLinkTarget);
+        .on("mouseout", function () {
+            hoverInfoLabel.style("opacity", config.node.hoverLabel.opacityOnNode);
+        });
 
     return links;
 }
 
 /**
+ * Show the link information when the user hovers over a link
+ * @param link the link to get the information of
+ * @param infoLabel the label to display the information in
+ */
+function mouseInEdge(link, infoLabel) {
+    infoLabel.style("opacity", config.node.hoverLabel.opacityOnEdge);
+    infoLabel.style("display", "block");
+
+    infoLabel.html(
+        "<table>" +
+        "<tr>" +
+        "<td class=\"metric\">Uploaded</td>" +
```

Comment:
I'd suggest renaming Uploaded and Downloaded by Uploaded by X and Uploaded by Y. This makes it independent from which node it is viewed and removes ambiguity